x-pack/legacy/plugins/monitoring/public/views/loading/index.js

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { render, unmountComponentAtNode } from 'react-dom';
+import { PageLoading } from 'plugins/monitoring/components';
+import uiRoutes from 'ui/routes';
+import { I18nContext } from 'ui/i18n';
+import template from './index.html';
+import { setAngularState, toggleSetupMode, getSetupModeState, initSetupModeState } from '../../lib/setup_mode';
+
+const REACT_DOM_ID = 'monitoringLoadingReactApp';
+
+uiRoutes
+  .when('/loading', {
+    template,
+    controller: class {
+      constructor($injector, $scope) {
+        const monitoringClusters = $injector.get('monitoringClusters');
+        const kbnUrl = $injector.get('kbnUrl');
+
+        setAngularState($scope, $injector);
+        initSetupModeState();
```

Comment:
Ref **A**: I know this is more from the previous PR, but is there any reason why we need to set state first and then use it? Maybe some kind of static singleton (where we don't keep state) approach would work better here eg:
```javascript
initSetupModeState($scope, $injector);
```
Do that for all exported handlers in `setup_mode.js` that depend on 'em, and remove `setAngularState` all together.

Might be nitpicking though

Reply:
Great suggestion! I'll change it